src/global_data.rs

```diff
@@ -0,0 +1,136 @@
+use crate::process_info::TxContext;
+use std::{
+    collections::HashMap,
+    ffi::c_void,
+    pin::Pin,
+    sync::{Mutex, MutexGuard},
+};
+
+lazy_static! {
+    static ref GLOBAL_DATA: Pin<Box<Mutex<GlobalData>>> = Pin::new(Box::default());
+}
+static mut GLOBAL_DATA_PTR: *mut Mutex<GlobalData> = std::ptr::null_mut();
+
+#[derive(Default, PartialEq, Eq, Clone, Hash, Debug)]
+pub struct TxID(u64);
+impl From<u64> for TxID {
+    fn from(value: u64) -> Self {
+        Self(value)
+    }
+}
+impl From<TxID> for u64 {
+    fn from(value: TxID) -> Self {
+        value.0
+    }
+}
+impl TxID {
+    fn next(&mut self) -> Self {
```

Comment:
Why do you modify the original data here, but clone a copy of the data as the return value at the same time?

Reply:
In many places, the `next` step is `HashMap::insert`. This way, the return is more convenient to use.